pkg/imgutil/snapshotter.go

```diff
@@ -77,9 +80,18 @@ func (rs *remoteSnapshotterOpts) isRemote() bool {
 
 func (rs *remoteSnapshotterOpts) apply(config *pull.Config, ref string) {
 	h := ctdsnapshotters.AppendInfoHandlerWrapper(ref)
+
+	if rs.snapshotter == "soci" {
+		h = socisource.AppendDefaultLabelsHandlerWrapper(
+			"", ctdsnapshotters.AppendInfoHandlerWrapper(ref))
```

Comment:
Can we implement this by extending `extraLabels`, instead of adding soci exception here?

Reply:
I changed implementation so soci uses extraLabels functionality now.